waddrmgr/manager.go

```diff
@@ -1858,7 +2170,7 @@ func Open(namespace walletdb.Namespace, pubPassphrase []byte, chainParams *chain
 	}
 
 	// Upgrade the manager to the latest version as needed.
-	if err := upgradeManager(namespace); err != nil {
+	if err := upgradeManager(namespace, pubPassphrase, config.ObtainSeed, config.ObtainPrivatePass); err != nil {
```

Comment:
This would panic if a nil config (which is valid) is passed.  I'd move the section below which sets the default config above this line and change the `upgradeManager` signature to take the entire config struct instead of the individual entries.